### PR TITLE
Replace ActiveFedora::Base.find

### PR DESCRIPTION
**Pull request by laritakr**
references https://github.com/samvera/hyrax/issues/3790

Replace the majority of direct calls to ActiveFedora::Base.find with calls to Valkyrie query service find_by_alternate_identifier.

Note: Does not address following cases:
* find is using a batch of ids rather than a single id
* nesting index adapter

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by laritakr** [comment]
Fixes two randomly failing specs from #3038. They needed additional time after the save so the content could become available. Commented them out to retain the reference to the issue in case their passing was a fluke. Comments can be removed in a later PR if they continue to pass.

File: app/services/hyrax/persist_directly_contained_output_file_service.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def self.call(content, directives)
     def self.retrieve_file_set(directives)
       uri = URI(directives.fetch(:url))
       raise ArgumentError, "#{uri} is not an http(s) uri" unless uri.is_a?(URI::HTTP)
-      ActiveFedora::Base.find(ActiveFedora::Base.uri_to_id(uri.to_s))
+      Valkyrie.config.metadata_adapter.query_service.find_by_alternate_identifier(alternate_identifier: ActiveFedora::Base.uri_to_id(uri.to_s), use_valkyrie: false)
```
Comment (jeremyf) [inline app/services/hyrax/persist_directly_contained_output_file_service.rb]: Do we need to cast the URI? Can we use `uri` instead of `ActiveFedora::Base.uri_to_id(uri.to_s)`?

Comment (laritakr) [reply app/services/hyrax/persist_directly_contained_output_file_service.rb]: uri_to_id is being addressed by another issue so I left this as is.

Comment (laritakr) [reply app/services/hyrax/persist_directly_contained_output_file_service.rb]: https://github.com/samvera/hyrax/issues/3821

File: lib/wings/valkyrie/query_service.rb [hunk 2]
```diff
@@ -76,12 +76,19 @@ def find_many_by_ids(ids:)
         resources
       end
 
-      def find_by_alternate_identifier(alternate_identifier:)
+      # Find a record using an alternate ID, and map it to a Valkyrie Resource
+      # @param [Valkyrie::ID, String] id
+      # @param [boolean] optionally return ActiveFedora object/errors
+      # @return [Valkyrie::Resource]
+      # @raise [Valkyrie::Persistence::ObjectNotFoundError]
```
Comment (elrayle) [inline lib/wings/valkyrie/query_service.rb]: Nice adding in the documentation!

File: spec/features/create_work_spec.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-RSpec.describe 'Creating a new Work', :js, :workflow do
+RSpec.describe 'Creating a new Work', :js, :workflow, :clean_repo do
```
Comment (elrayle) [inline spec/features/create_work_spec.rb]: Did something in the PR cause the need for `:clean_repo`?

File: spec/features/static_pages_spec.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
-RSpec.describe "The static pages" do
+RSpec.describe "The static pages", :clean_repo do
```
Comment (elrayle) [inline spec/features/static_pages_spec.rb]: Did something in the PR cause the need for `:clean_repo`?

File: spec/features/work_show_spec.rb [hunk 5]
```diff
@@ -55,7 +55,7 @@
     end
 
     it "allows adding work to a collection", clean_repo: true, js: true do
-      optional 'ability to get capybara to find css select2-result (see Issue #3038)' if ci_build?
+      # optional 'ability to get capybara to find css select2-result (see Issue #3038)' if ci_build?
```
Comment (elrayle) [inline spec/features/work_show_spec.rb]: Can you elaborate on why this change and the addition of `sleep 5` below are needed?

File: spec/features/work_show_spec.rb [hunk 6]
```diff
@@ -107,12 +108,13 @@
...
```
Comment (elrayle) [inline spec/features/work_show_spec.rb]: same question here

File: spec/wings/valkyrie/query_service_spec.rb [hunk 7]
```diff
@@ -152,6 +152,24 @@ class Image < ActiveFedora::Base
       expect(found.id).to eq resource.id
       expect(found).to be_persisted
     end
+
+    context 'when use_valkyrie: false' do
+      it 'returns an ActiveFedora object' do
+        resource = resource_class.new
```
Comment (elrayle) [inline spec/wings/valkyrie/query_service_spec.rb]: Any reason not to set `id` outside the test using...
```
let(:id) do
  resource.alternate_ids = [Valkyrie::ID.new('p9s0xfj')]
  resource = persister.save(resource: resource)
  resource.alternate_ids.first
end
```

File: spec/features/static_pages_spec.rb (hunk 4 above)
Comment (laritakr) [reply spec/features/static_pages_spec.rb]: No this has been a randomly failing spec for me. It was failing due to some missing thumbnail in the "recent objects" list due to an object that didn't get created fully elsewhere.

File: spec/features/work_show_spec.rb (hunk 5 above)
Comment (laritakr) [reply spec/features/work_show_spec.rb]: After a save is issued, there can be some delay as it processes before the object is available for the subsequent commands. If you debug, it slows things down and works, but when run without debug, it fails. Sleep allows the page to be fully available for the subsequent tests.

File: spec/features/work_show_spec.rb (hunk 6 above)
Comment (laritakr) [reply spec/features/work_show_spec.rb]: (same answer) :)

File: spec/features/create_work_spec.rb (hunk 3 above)
Comment (laritakr) [reply spec/features/create_work_spec.rb]: If this fails once, then it continues to fail. The work gets created with the same title as previous attempts, so the page has multiple works with the same title, and a later find fails.

File: spec/features/static_pages_spec.rb (hunk 4 above)
Comment (laritakr) [reply spec/features/static_pages_spec.rb]: (Randomly failing both locally and in CI)

File: spec/features/work_show_spec.rb (hunk 5 above)
Comment (no-reply) [reply spec/features/work_show_spec.rb]: okay to just remove this line, rather than commenting?

File: spec/wings/valkyrie/query_service_spec.rb (hunk 7 above)
Comment (laritakr) [reply spec/wings/valkyrie/query_service_spec.rb]: I copied the code from an earlier spec and changed it for the "use_valkyrie" parameter. I'm not positive, but I believe that I split it out from the code elsewhere to placate Rubocop who thought line 163 was too long when it included `found = query_service.find_by_alternate_identifier(alternate_identifier: resource.alternate_ids.first, use_valkyrie: false)'. Naming it id is probably a bit unclear, given all of the other ids referenced in that spec.

File: spec/features/work_show_spec.rb (hunk 5 above)
Comment (laritakr) [reply spec/features/work_show_spec.rb]: I left the comment temporarily, to keep the reference to the related issue until I was sure it fixed the problem.  I intended to remove it later, but if this PR has other changes needed, I'll remove it here.

File: spec/features/static_pages_spec.rb (hunk 4 above)
Comment (elrayle) [reply spec/features/static_pages_spec.rb]: Thanks for the clarification.  And nice to see changes that make the tests more stable.

File: spec/wings/valkyrie/query_service_spec.rb (hunk 7 above)
Comment (elrayle) [reply spec/wings/valkyrie/query_service_spec.rb]: No problem.  This is more of a style thing.  I generally try to put setup code that isn't part of what is being tested outside the test itself.  Makes what is being tested easier to see at a glance.

File: spec/features/work_show_spec.rb (hunk 5 above)
Comment (elrayle) [reply spec/features/work_show_spec.rb]: My worry would be that these kinds of comments can end up being out of sight, out of mind and never get deleted until someone stumbles across it.  And that person may not be sure it is ok to remove.  

Perhps you could add a comment referencing the issue near the `sleep` statement.  

_Caveat: I'm not familiar with `optional` in rspec tests and a quick google search did not enlighten me.  So my suggestion may not make sense if I guessed its function incorrectly._

Something like...
```
# This attempts to address issue #3038.  Previously, this test was marked optional.

Comment (laritakr) [reply spec/features/work_show_spec.rb]: I could, but honestly, I intend to remove the comment entirely in my next PR. I just wanted to leave it when I pushed up the code, to make sure it fixed it on Travis as well as locally, so it would be easier to find if it didn't work. Is it worth removing it or adding a different comment in this PR? or should i just do it as part of my next one?

Comment (elrayle) [reply spec/features/work_show_spec.rb]: That seems fine if you are doing it in the next PR.
